modules/citrus-http/src/main/java/com/consol/citrus/http/message/CookieConverter.java

```diff
@@ -54,6 +63,45 @@
         return cookies.toArray(new Cookie[0]);
     }
 
+    /**
+     * Converts a given cookie into a HTTP conform cookie String
+     * @param cookie the cookie to convert
+     * @return The cookie string representation of the given cookie
+     */
+    String getCookieString(Cookie cookie) {
+        StringBuilder builder = new StringBuilder();
+
+        builder.append(cookie.getName());
+        builder.append("=");
+        builder.append(cookie.getValue());
+
+        if (cookie.getVersion() > 0) {
+            builder.append(";" + VERSION + "=").append(cookie.getVersion());
+        }
+
+        if (StringUtils.hasText(cookie.getPath())) {
+            builder.append(";" + PATH + "=").append(cookie.getPath());
+        }
+
+        if (StringUtils.hasText(cookie.getDomain())) {
+            builder.append(";" + DOMAIN + "=").append(cookie.getDomain());
+        }
+
+        if (cookie.getMaxAge() > 0) {
+            builder.append(";" + MAX_AGE + "=").append(cookie.getMaxAge());
+        }
+
+        if (StringUtils.hasText(cookie.getComment())) {
+            builder.append(";" + COMMENT + "=").append(cookie.getComment());
+        }
+
+        if (cookie.getSecure()) {
+            builder.append(";" + SECURE);
+        }
+
```

Comment:
What about HttpOnly flag?

Reply:
Good point! `HttpOnly` was missing the whole time.